datafusion/functions/src/string/chr.rs

```diff
@@ -47,22 +47,14 @@ pub fn chr(args: &[ArrayRef]) -> Result<ArrayRef> {
     for integer in integer_array {
         match integer {
             Some(integer) => {
-                if integer == 0 {
-                    return exec_err!("null character not permitted.");
-                } else if integer < 0 {
-                    return exec_err!("negative input not permitted.");
-                } else {
-                    match core::char::from_u32(integer as u32) {
-                        Some(c) => {
-                            builder.append_value(c.encode_utf8(&mut buf));
-                        }
-                        None => {
-                            return exec_err!(
-                                "requested character too large for encoding."
-                            );
-                        }
+                if integer >= 0 && integer <= u32::MAX as i64 {
```

Comment:
I think you can probably express this more succiently using try_from 

Something like 
```suggestion
                let Ok(integer) = u32::try_from(integer) else {
                  return exec_err!("invalid Unicode scalar value: {integer}");
                }              
```

But I don't think it is necessary

Reply:
That's better indeed. Code adjusted.